packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -0,0 +1,146 @@
+import { renderPlaygroundPage } from '@apollographql/graphql-playground-html';
+import { Accepts } from 'accepts';
+import {
+  ApolloServerBase,
+  PlaygroundRenderPageOptions,
+} from 'apollo-server-core';
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { IncomingMessage, OutgoingMessage, Server } from 'http';
+import { processRequest as processFileUploads } from '@apollographql/apollo-upload-server';
+import { graphqlFastify } from './fastifyApollo';
+
+const fastJson = require('fast-json-stringify');
+
+export interface ServerRegistration {
+  path?: string;
+  cors?: object | boolean;
+  onHealthCheck?: (req: FastifyRequest<IncomingMessage>) => Promise<any>;
+  disableHealthCheck?: boolean;
+}
+
+const stringifyHealthCheck = fastJson({
+  type: 'object',
+  properties: {
+    status: {
+      type: 'string',
+    },
+  },
+});
+
+export class ApolloServer extends ApolloServerBase {
+  protected supportsSubscriptions(): boolean {
+    return true;
+  }
+
+  protected supportsUploads(): boolean {
+    return true;
+  }
+
+  public createHandler({
+    path,
+    cors,
+    disableHealthCheck,
+    onHealthCheck,
+  }: ServerRegistration = {}) {
+    this.graphqlPath = path ? path : '/graphql';
+    const promiseWillStart = this.willStart();
+
+    return async (
+      app: FastifyInstance<Server, IncomingMessage, OutgoingMessage>,
+    ) => {
+      await promiseWillStart;
+
+      if (!disableHealthCheck) {
+        app.get('/.well-known/apollo/server-health', async (req, res) => {
+          // Response follows https://tools.ietf.org/html/draft-inadarei-api-health-check-01
+          res.type('application/health+json');
+
+          if (onHealthCheck) {
+            try {
+              await onHealthCheck(req);
+              res.send(stringifyHealthCheck({ status: 'pass' }));
+            } catch (e) {
+              res.status(503).send(stringifyHealthCheck({ status: 'fail' }));
+            }
+          } else {
+            res.send(stringifyHealthCheck({ status: 'pass' }));
+          }
+        });
+      }
+
+      app.register(
+        async instance => {
+          instance.register(require('fastify-accepts'));
+
+          if (cors === true) {
+            instance.register(require('fastify-cors'));
+          } else if (cors !== false) {
+            instance.register(require('fastify-cors'), cors);
+          }
+
+          instance.setNotFoundHandler((_request, reply) => {
+            reply.code(405);
+            reply.header('allow', 'GET, POST');
+            reply.send();
+          });
+
+          instance.addContentTypeParser(
+            'multipart',
+            async (request: IncomingMessage) =>
+              processFileUploads(request, this.uploadsConfig),
```

Comment:
Some slight tweaks are necessary to account for #2054.

Reply:
https://github.com/apollographql/apollo-server/pull/1971/commits/19d96320ddf295ff4b9d24d8359eb17490b9855d